src/test/java/com/synopsys/integration/jenkins/scan/factory/ScanParametersFactoryTest.java

```diff
@@ -0,0 +1,185 @@
+    private TaskListener listenerMock;
+    private FilePath workspace;
+    private EnvVars envVarsMock;
+    private SecurityScanStep securityScanStep;
+
+    @BeforeEach
+    public void setUp() {
+        workspace = new FilePath(new File(System.getProperty("user.home")));
+        listenerMock = Mockito.mock(TaskListener.class);
+        envVarsMock = Mockito.mock(EnvVars.class);
+        securityScanStep = new SecurityScanStep();
+        Mockito.when(listenerMock.getLogger()).thenReturn(Mockito.mock(PrintStream.class));
+    }
+
+    @Test
+    public void preparePipelineParametersMapTest() throws AbortException, PluginExceptionHandler {
+        Map<String, Object> globalConfigValues = new HashMap<>();
+        ScanParametersFactory scanParametersFactory = new ScanParametersFactory(envVarsMock, workspace);
+
+        securityScanStep.setProduct("BLACKDUCK");
+        securityScanStep.setBitbucket_token("FAKETOKEN");
+        globalConfigValues.put(ApplicationConstants.BLACKDUCK_URL_KEY, "https://fake-blackduck.url");
+        globalConfigValues.put(ApplicationConstants.BLACKDUCK_TOKEN_KEY, "fake-blackduck-token");
+        globalConfigValues.put(ApplicationConstants.SYNOPSYS_BRIDGE_INSTALL_DIRECTORY, "/fake/path");
+
+        Map<String, Object> result = scanParametersFactory.preparePipelineParametersMap(securityScanStep,
+                globalConfigValues, workspace, listenerMock);
+
+        assertEquals(5, result.size());
+        assertEquals("BLACKDUCK", result.get(ApplicationConstants.PRODUCT_KEY));
+        assertEquals("fake-blackduck-token", result.get(ApplicationConstants.BLACKDUCK_TOKEN_KEY));
+        assertEquals("/fake/path", result.get(ApplicationConstants.SYNOPSYS_BRIDGE_INSTALL_DIRECTORY));
+        assertEquals("FAKETOKEN", result.get(ApplicationConstants.BITBUCKET_TOKEN_KEY));
+
+        securityScanStep.setProduct("invalid-product");
+
+        assertThrows(PluginExceptionHandler.class, () -> scanParametersFactory.preparePipelineParametersMap(securityScanStep,
+                globalConfigValues, workspace, listenerMock));
+    }
+
+    @Test
+    public void prepareBlackDuckParametersMapTest() {
+        securityScanStep.setBlackduck_url("https://fake.blackduck-url");
+        securityScanStep.setBlackduck_token("fake-token");
+        securityScanStep.setBlackduck_install_directory("/fake/path");
+        securityScanStep.setBlackduck_scan_full(true);
+        securityScanStep.setBlackduck_automation_prcomment(true);
+        securityScanStep.setBlackduck_download_url("https://fake.blackduck-download-url");
+        securityScanStep.setBlackduck_scan_failure_severities("MAJOR");
+
+        Map<String, Object> blackDuckParametersMap = ScanParametersFactory.prepareBlackDuckParametersMap(securityScanStep);
+
+        assertEquals(7, blackDuckParametersMap.size());
+        assertEquals("https://fake.blackduck-url", blackDuckParametersMap.get(ApplicationConstants.BLACKDUCK_URL_KEY));
```

Comment:
We are using the same string values in multiple places. I think we should define  these values in  variables and then access those from the variables. 
For example, we are repeating this `"https://fake.blackduck-url"`